src/tscore/CMakeLists.txt

```diff
@@ -0,0 +1,151 @@
+#######################
+#
+#  Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+#  agreements.  See the NOTICE file distributed with this work for additional information regarding
+#  copyright ownership.  The ASF licenses this file to you under the Apache License, Version 2.0
+#  (the "License"); you may not use this file except in compliance with the License.  You may obtain
+#  a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software distributed under the License
+#  is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing permissions and limitations under
+#  the License.
+#
+#######################
+
+
+
+# This executable generates the parse rules that are included in ParseRules.cc
+# Add custom commands and dependencies to make sure this gets build and run before compiling libtscore
+add_executable(CompileParseRules CompileParseRules.cc)
+
+add_custom_command(
+        OUTPUT ParseRulesCType ParseRulesCTypeToUpper ParseRulesCTypeToLower
+        COMMAND CompileParseRules
+        COMMENT "Generating compile parse rules"
+)
+
+add_custom_target(ParseRules ALL DEPENDS ParseRulesCType ParseRulesCTypeToUpper ParseRulesCTypeToLower)
+
+add_library(tscore SHARED
+        AcidPtr.cc
+        Arena.cc
+        ArgParser.cc
+        BaseLogFile.cc
+        BufferWriterFormat.cc
+        ConsistentHash.cc
+        ContFlags.cc
+        CryptoHash.cc
+        DbgCtl.cc
+        Diags.cc
+        Errata.cc
+        EventNotify.cc
+        Extendible.cc
+        HKDF_openssl.cc
```

Comment:
Can we support boringssl and openssl3? It doesn't have to be done on this PR, but I want to make sure it won't require a drastic change.

Reply:
Yes, just have to add the options and then conditionally add the correct source